File: functions/src/models/question_of_the_day.ts
import mongoose from 'mongoose'

const questionOfTheDaySchema = new mongoose.Schema({
  question: String,
  date: {
    type: String,
    unique: true,
  },
  assertions: [String],
  reponse: String,
})

const QuestionOfTheDay = mongoose.model(
  'questions_of_the_days',
  questionOfTheDaySchema,
)

export { questionOfTheDaySchema, QuestionOfTheDay }

File: lib/data/network/entities/question_of_the_day.dart
import 'package:json_annotation/json_annotation.dart';

part 'question_of_the_day.g.dart';

@JsonSerializable()
class QuestionOfTheDayData {
  @JsonKey(name: '_id')
  final String id;
  final String question;
  final String date;
  final String reponse;
  final List<String> assertions;

  QuestionOfTheDayData({
    required this.id,
    required this.question,
    required this.assertions,
    required this.date,
    required this.reponse,
  });

  factory QuestionOfTheDayData.fromJson(Map<String, dynamic> json) => _$QuestionOfTheDayDataFromJson(json);
  Map<String, dynamic> toJson() => _$QuestionOfTheDayDataToJson(this);
}
